The `rosbag2_performance_benchmarking` fails to run with `test_benchmark_producers` config

## Description
The `rosbag2_performance_benchmarking` fails at the end due to a non-zero exit code from the `ros2 bag record` process when using `test_benchmark_producers` config.

### Expected Behavior
The `rosbag2_performance_benchmarking` shall not fail and exit correctly without error codes.

### Actual Behavior
The `rosbag2_performance_benchmarking` fails on finish with an error code 
```bash
[ros2-5] child process pid 47023 exit status already read:  will report returncode 255
```
when using `test_benchmark_producers` config.

## To Reproduce
Steps to reproduce the behavior, e.g.
1. colcon build --cmake-args -DBUILD_ROSBAG2_BENCHMARKS=1 --packages-up-to rosbag2_performance_benchmarking
2. colcon test --packages-select rosbag2_performance_benchmarking
3. Scroll down to the first failure messages in the log
4. Logs print error: 
```bash
1: [ros2-5] [ros2-1] [INFO] [1752287756.002961670] [rosbag2_recorder]: Subscribed to topic '/rosbag2_performance_benchmarking_node/imu_1'
1: [ros2-5] [ros2-1] [INFO] [1752287756.003589639] [rosbag2_recorder]: Subscribed to topic '/rosbag2_performance_benchmarking_node/radar_2'
1: [ros2-5] [ros2-1] [INFO] [1752287756.004201333] [rosbag2_recorder]: Subscribed to topic '/rosbag2_performance_benchmarking_node/ultrasonic_8'
1: [ros2-5] [benchmark_publishers-2] [INFO] [1752287766.588846993] [rosbag2_performance_benchmarking_node]: Producer threads finished
1: [ros2-5] [benchmark_publishers-2] [INFO] [1752287767.589370670] [rosbag2_performance_benchmarking_node]: Benchmark terminated
1: [ros2-5] [INFO] [benchmark_publishers-2]: process has finished cleanly [pid 116547]
1: [ros2-5] child process pid 116528 exit status already read:  will report returncode 255
1: [ros2-5] [ros2-1] [INFO] [1752287767.806457122] [rosbag2_recorder]: Pausing recording.
1: [ros2-5] [ros2-1] [INFO] [1752287767.806519497] [rosbag2_cpp]: Writing remaining messages from cache to the bag. It may take a while
1: [ros2-5] [ros2-1] [INFO] [1752287767.816066854] [rosbag2_recorder]: Event publisher thread: Exiting
1: [ros2-5] [ros2-1] [INFO] [1752287767.816095412] [rosbag2_recorder]: Recording stopped
1: [ros2-5] [ros2-1] [INFO] [1752287767.819095307] [rclcpp]: signal_handler(signum=2)
1: [ros2-5] [ERROR] [ros2-1]: process has died [pid 116528, exit code 255, cmd 'ros2 bag record -e \/.*_benchmarking_node\/.* -s mcap --max-cache-size 100000000 --compression-queue-size 1 -o /tmpfs/rosbag2_performance_test/test_benchmark_producers_automotive_transport_2025-07-11_19-35-55/run_0_mcap_100000000_default_compression_1_0_default_config_0'].
1: [ros2-5] [INFO] [launch.user]: Rosbag2 record error. Shutting down benchmark. Return code = 255
1: [ros2-5] [INFO] [results_writer-3]: process started with pid [116593]
1: [ros2-5] [INFO] [results_writer-3]: sending signal 'SIGINT' to process[results_writer-3]
1: [ros2-5] [ERROR] [results_writer-3]: process has died [pid 116593, exit code -2, cmd '/home/morlov/ros2_jazzy/install/rosbag2_performance_benchmarking/lib/rosbag2_performance_benchmarking/results_writer --ros-args -r __node:=rosbag2_performance_benchmarking_node --params-file /home/morlov/ros2_jazzy/install/rosbag2_performance_benchmarking/share/rosbag2_performance_benchmarking/config/producers/automotive.yaml --params-file /tmp/launch_params_ciy9696t --params-file /tmp/launch_params_wn4wstuu --params-file /tmp/launch_params_4fxj63mx --params-file /tmp/launch_params_vu8rdtja --params-file /tmp/launch_params_0_79_fwn --params-file /tmp/launch_params_rap3wqbn --params-file /tmp/launch_params_d5azodpb --params-file /tmp/launch_params_tzofck4n --params-file /tmp/launch_params_jrm3xqv_ --params-file /tmp/launch_params_kg145gc8'].
1: [ros2-5] [INFO] [launch.user]: Results writer error. Shutting down benchmark. Return code = -2
1: [INFO] [ros2-5]: process has finished cleanly [pid 116525]
1: ok
```

## System (please complete the following information)
 - OS: Ubuntu 22.04 LTS
 - ROS 2 Distro: Rolling, Kilted, Jazzy
 - Install Method: Build from sources
 - Version: latest releases. (Rolling 0.33.0)


## Additional context
We haven't observed this issue before and didn't make much changes in the `rosbag2_performance_benchmarking` package recently.
However, the tests from the `rosbag2_performance_benchmarking` are not running on CI.

--- a/rosbag2_performance/rosbag2_performance_benchmarking/launch/benchmark_launch.py
+++ b/rosbag2_performance/rosbag2_performance_benchmarking/launch/benchmark_launch.py
@@ -51,6 +51,7 @@ import pathlib
 import shutil
 import signal
 import sys
+import time
 
 from ament_index_python import get_package_share_directory
 import launch
@@ -267,9 +268,19 @@ def _producer_node_exited(event, context):
             ]
         _recorder_cpu_usage = _rosbag_process.cpu_percent()
         os.kill(_rosbag_pid, signal.SIGINT)
-        # Wait for rosbag2 process to exit for 10 seconds
-        rosbag_return_code = _rosbag_process.wait(10)
         _rosbag_pid = None
+        # Wait for rosbag2 process to exit for 10 seconds
+        timeout = 10
+        start_time = time.time()
+        rosbag_return_code = None
+        # Note: we cant use _rosbag_process.wait(10) because wait shall be called only once.
+        # We will have to call wait one more time in the test_proc_terminates(self, last_benchmark)
+        while time.time() - start_time < timeout:
+            if not _rosbag_process.is_running():
+                rosbag_return_code = _rosbag_process.returncode()
+                break
+            time.sleep(0.1)
+
         if rosbag_return_code is not None and rosbag_return_code != 0:
             return [
                 launch.actions.LogInfo(msg='Rosbag2 record error. Shutting down benchmark. '

